src/index.ts

```diff
@@ -1,43 +1,66 @@
 import { v4 as uuidv4 } from 'uuid';
 
 type AlgorithmFunction = () => string;
+interface MemoizationEntry {
+  uri: string;
+  uuidAlgorithm: string;
+}
 
 const algorithms: Record<string, AlgorithmFunction> = {
   uuid4: uuidv4,
 };
 
 function replacingAlgorithm(
-  memoizationTable: Record<string, string>,
+  memoizationTable: Record<string, MemoizationEntry>,
   _: unknown,
   uriBase: string,
   uuidAlgorithm: string,
   originalUuid: string,
 ) {
-  if (memoizationTable[`${uriBase}${originalUuid}`]) {
-    return memoizationTable[`${uriBase}${originalUuid}`];
+  const memoizationEntry = memoizationTable[`${uriBase}${originalUuid}`];
+  if (memoizationEntry) {
+    if (memoizationEntry.uuidAlgorithm !== uuidAlgorithm) {
+      console.warn(
+        `You have 2 entries with the same uri base and uuid, but different algorithm, this is probably and error so the result will be the same uuid in both entries. ${uriBase}--ref-<${uuidAlgorithm}>-${originalUuid} and ${uriBase}--ref-<${memoizationEntry.uuidAlgorithm}>-${originalUuid}`,
+      );
+    }
+    return memoizationTable[`${uriBase}${originalUuid}`].uri;
   } else {
+    if (!algorithms[uuidAlgorithm]) {
+      console.warn(`${uuidAlgorithm} was not recognised, using uuid4 instead`);
+    }
     const algorithm = algorithms[uuidAlgorithm] || uuidv4;
     const finalUri = `"${uriBase}${algorithm()}"`;
-    memoizationTable[`${uriBase}${originalUuid}`] = finalUri;
+    memoizationTable[`${uriBase}${originalUuid}`] = {
+      uri: finalUri,
+      uuidAlgorithm,
+    };
     return finalUri;
   }
 }
 
-// We disable typescript linting here as the curry function should indeed accept any function and any arguments
-// eslint-disable-next-line  @typescript-eslint/no-explicit-any, @typescript-eslint/ban-types
-function currying(fn: Function, ...args: any[]) {
-  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
-  return (..._arg: any) => {
-    return fn(...args, ..._arg);
+function curryReplacingAlgorithm(
+  memoizationTable: Record<string, MemoizationEntry>,
+) {
+  return (
+    _: unknown,
+    uriBase: string,
+    uuidAlgorithm: string,
+    originalUuid: string,
+  ) => {
+    return replacingAlgorithm(
+      memoizationTable,
+      _,
```

Comment:
There's no reason for this to be here. That was what I intended by changing the way the currying was done.